### PR TITLE
MNTOR-2599: delete breach resolution object from token

### PR DESCRIPTION
**Pull request by mansaj**
# References: 
Jira: MNTOR-2599

# Description
The session token contains breach_resolution object, which can get way too big for some users and people are seeing 431 HTTP error. 

This PR filters that object out of the session token

I've scanned the code base and it doesn't seem to be used by anything. We always fetch the subscriber object from DB directly in API calls

# Screenshot (if applicable)

# How to test
* Bulk resolve a lot of breaches
* Make sure we aren't getting `431 Header too large` error anymore

# Checklist (Definition of Done)
- [x] Commits in this PR are minimal and [have descriptive commit messages](https://chris.beams.io/posts/git-commit/).
- [x] I've added or updated the relevant sections in readme and/or code comments
- [x] All acceptance criteria are met.
- [x] Jira ticket has been updated (if needed) to match changes made during the development process.
- [x] Jira ticket has been updated (if needed) with suggestions for QA when this PR is deployed to stage.

### CODE REVIEW

**Review by flozia — APPROVED** [review]
Nice find!

File: src/app/api/utils/auth.ts [hunk 1]
```diff
@@ -88,6 +88,7 @@ export const authOptions: AuthOptions = {
         //       multiple Accounts at multiple providers).
         const email = profile.email;
         const existingUser = await getSubscriberByEmail(email);
+        delete existingUser.breach_resolution;
```
Comment (Vinnl) [inline src/app/api/utils/auth.ts:91]:
```suggestion
        // MNTOR-2599 The breach_resolution object can get pretty big,
        // causing the session token cookie to balloon in size,
        // eventually resulting in a 400 Bad Request due to headers being too large.
        delete existingUser.breach_resolution;
```

File: src/app/api/utils/auth.ts [hunk 2]
```diff
@@ -98,6 +99,8 @@ export const authOptions: AuthOptions = {
               account.refresh_token,
               JSON.stringify(profile),
             );
+            // MNTOR-2599: delete breach_resolution in token
```
Comment (Vinnl) [inline src/app/api/utils/auth.ts:102]:
```suggestion          
            // MNTOR-2599 The breach_resolution object can get pretty big,
            // causing the session token cookie to balloon in size,
            // eventually resulting in a 400 Bad Request due to headers being too large.
```

**Review by Vinnl — APPROVED** [review]
Great sleuthing! Would be helpful to add a comment for future readers to understand why we're doing this though :)

**Review by rhelmer — CHANGES_REQUESTED** [review]
Is the token the right place to store this breach resolution information? I'm not sure why it isn't in the database, or in the body of an HTTP request? What is the use case for having it be part of the session?

**Comment by mansaj** [comment]
> Is the token the right place to store this breach resolution information? I'm not sure why it isn't in the database, or in the body of an HTTP request? What is the use case for having it be part of the session?

Well.. This PR takes that object *out* of the session token.. I think it was in session before just for convenience? Or the auth code was copy pasted from the old auth code, which included everything in `subscriber` in the session token.

File: src/app/api/utils/auth.ts [hunk 3]
```diff
@@ -90,14 +90,24 @@ export const authOptions: AuthOptions = {
...
 
         if (existingUser) {
+          // MNTOR-2599 The breach_resolution object can get pretty big,
+          // causing the session token cookie to balloon in size,
+          // eventually resulting in a 400 Bad Request due to headers being too large.
+          delete existingUser.breach_resolution;
           token.subscriber = existingUser;
           if (account.access_token && account.refresh_token) {
+            console.log({ existingUser });
+            console.log({ profile });
```
Comment (flozia) [inline src/app/api/utils/auth.ts:100]:
```suggestion
```

Comment (pdehaan) [inline src/app/api/utils/auth.ts:100]: ✂️

**Pull request merged**